core/node/libp2p/rcmgr_logging.go

```diff
@@ -47,11 +48,23 @@ func (n *loggingResourceManager) start(ctx context.Context) {
 			case <-ticker.C:
 				n.mut.Lock()
 				errs := n.limitExceededErrs
-				n.limitExceededErrs = 0
-				n.mut.Unlock()
-				if errs != 0 {
-					n.logger.Warnf("Resource limits were exceeded %d times, consider inspecting logs and raising the resource manager limits.", errs)
+				n.limitExceededErrs = make(map[string]int)
+
+				for e, count := range errs {
+					n.previousErrors = true
+					n.logger.Errorf("Resource limits were exceeded %d times with error %q.", count, e)
+				}
+
+				if len(errs) != 0 {
+					n.logger.Errorf("Consider inspecting logs and raising the resource manager limits. Documentation: https://github.com/ipfs/kubo/blob/master/docs/config.md#swarmresourcemgr")
+				}
+
+				if len(errs) == 0 && n.previousErrors {
+					n.previousErrors = false
+					n.logger.Errorf("Resrouce limits are no longer being exceeded.")
```

Comment:
This is not an error so I don't think it deserves the error severity, maybe make it info? (Yes that means it won't show up in the logs by default :( but I think it's important to tag logs correctly.)

```suggestion
					n.logger.Infof("Resource limits are no longer being exceeded.")
```

Reply:
I'll remove it because it is not giving a lot of info to the user.